src/main/java/duke/command/Duke.java

```diff
@@ -0,0 +1,337 @@
+            doneCheck.printIntegerOnly();
+        } catch (NullPointerException e) {
+            doneCheck.printNoNull();
+        }
+    }
+
+    private static int addTaskToList(Scanner in, ArrayList<Task> taskList) {
+        String taskName;
+        do {
+            taskName = in.nextLine();
+            if (taskName.equals("stop")) {
+                return STOP_ADD;
+            }
+            taskList.add(new Task(taskName));
+            taskCount++;
+        } while (!taskName.equals("stop"));
+        System.out.println("Finished adding tasks!");
+        return ADD_SUCCESS;
+    }
+
+    private static int addTodoToList(Scanner in, ArrayList<Task> taskList) {
+        String todoName;
+        do {
+            todoName = in.nextLine();
+            if (todoName.equals("stop")) {
+                return STOP_ADD;
+            }
+            taskList.add(new Task(todoName, "todo"));
+            taskCount++;
+        } while (!todoName.equals("stop"));
+        System.out.println("Finished adding todos!");
+        return ADD_SUCCESS;
+    }
+
+    private static int addEventToList(Scanner in, ArrayList<Task> taskList) {
+        String eventName;
+        do {
+            eventName = in.nextLine();
+            if (eventName.equals("stop")) {
+                return STOP_ADD;
+            }
+            taskList.add(new Task(eventName, "event"));
+            taskCount++;
+        } while (!eventName.equals("stop"));
+        System.out.println("Finished adding todos!");
+        return ADD_SUCCESS;
+    }
+
+    private static void amendTaskDeadline(Scanner in, ArrayList<Task> taskList) {
+        DukeException deadlineCheck = new DukeException("deadlineCheck");
+        try {
+            String input;
+            do {
+                input = in.nextLine();
+                input = input.trim();
+                if (input.startsWith("stop")) {
+                    break;
+                }
+                if (!input.contains("/")) {
+                    deadlineCheck.printDeadlineFormatIssue();
+                    continue;
+                }
+                String[] separateInput = input.split("/", 2);
+                if (separateInput.length == 2) {
+                    //checks for singular, digit input
+                    if (!deadlineCheck.isIntegerInput(separateInput[0])) {
+                        deadlineCheck.printIntegerOnly();
+                        deadlineCheck.printDeadlineFormatIssue();
+                        continue;
+                    }
+                    int index = Integer.parseInt(separateInput[0].trim()) - 1;
+                    if (!deadlineCheck.inListRange(index, taskCount)) {
+                        deadlineCheck.printNotInRange(index);
+                        continue;
+                    }
+                    Task toChange = taskList.get(index);
+                    toChange.setDeadline(separateInput[1]);
+                    toChange.setToDo(true);
+                }
```

Comment:
From this line on there is heavy nesting. Try to restructure and see if there is a way for you to make the [happy path](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent) more prominent.